lib/IRGen/IRGenSIL.cpp

```diff
@@ -5014,6 +5013,10 @@ void IRGenSILFunction::setAllocatedAddressForBuffer(SILValue v,
 }
 
 void IRGenSILFunction::visitCopyAddrInst(swift::CopyAddrInst *i) {
+  CanSILFunctionType funcType = CurSILFn->getLoweredFunctionType();
```

Comment:
Can you push this down to the location where it's actually necessary, or refactor it away altogether? We're tyring to get rid of GenericContextScope, so I'd rather not see new usages appear

Reply:
I need this usage and this is the only place I can put it without a dirty hack: just like my comment above: need to set the generic context for usage down the call stack. This is the last location wherein I have that information before starting the outlined function creation

Reply:
I'm going to remove it soon so please find another way of expressing this.

Reply:
It is used in other passes that need it as well (such as large type API pass) - I am not sure removing it this late without discussion is such a good idea.

Reply:
It was put in as a hack and there are always better ways to express what it is doing.